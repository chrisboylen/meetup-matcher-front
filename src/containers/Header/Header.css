.navbar {
  align-items: center;
  background-color: #fff5f5c7;
  display: flex;
  height: 9rem;
  justify-content: space-between;
  padding: 0; }
  .navbar h1 {
    font-family: 'Paytone One', sans-serif;
    color: #ff0000;
    font-size: 3.5rem;
    margin: 0 0 0 5rem; }

.user-links {
  display: flex; }
  .user-links a {
    text-decoration: none;
    font-size: 1.5rem;
    color: black;
    cursor: pointer;
    padding: 3.6rem 2rem 3.6rem 2rem;
    align-self: center;
    margin: 0; }
    .user-links a button {
      background-color: transparent;
      border: 0;
      font-size: 1.3rem; }
  .user-links a:hover {
    background-color: black;
    color: #ff0000; }
  .user-links a:hover > * {
    color: #ff0000; }
