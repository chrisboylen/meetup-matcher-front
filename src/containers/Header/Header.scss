$bkgWhite: #fff5f5c7;
$mainRed: #ff0000;

.navbar {
  align-items: center;
  background-color: $bkgWhite;
  display: flex;
  height: 9rem;
  justify-content: space-between;
  padding: 0;
    .title-link {
      text-decoration: none;
    }
    h1 {
      font-family: 'Paytone One', sans-serif;
      color: $mainRed;
      font-size: 3.5rem;
      margin: 0 0 0 5rem;
    }
}
  
.user-links {
  display: flex;
    a {
      text-decoration: none;
      font-size: 1.5rem;
      color: black;
      cursor: pointer;
      padding: 3.6rem 2rem 3.6rem 2rem;
      align-self: center;
      margin: 0;
    }
    a:hover {
      background-color: black;
      color: $mainRed; 
    }
}
