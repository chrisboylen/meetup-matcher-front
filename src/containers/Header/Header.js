import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { logoutUser } from '../../actions';
import './Header.css';
import { auth } from '../../firebase/firebase';
import { logOutFirebase } from '../../firebase/auth';

export class Header extends Component {
  constructor () {
    super();
  }

  handleLogout = async () => {
    const { logout, history } = this.props;
    await logOutFirebase();
    await logout();
    history.push('/');
  };

  render () {
    return (
      <header className="navbar">
        <Link 
          className="title-link"
          to="/">
          <h1 className="full-txt">Meetup Matcher</h1>
          <h1 className="short-txt">MM</h1>
        </Link>
        <div className="user-links">
          <Link
            className="form-link login"
            to="/login">
              Login
          </Link>
          <Link
            className="form-link signup"
            to="/signup">
              Signup
          </Link>
          { auth.currentUser ?
            <Link
              className="form-link logout"
              to="" 
              onClick={this.handleLogout}>
                Log Out
            </Link> : ''
          }      
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  logout: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logoutUser())
});

export default withRouter(connect(null, mapDispatchToProps)(Header));