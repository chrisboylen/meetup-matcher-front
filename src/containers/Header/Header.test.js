import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';

describe('Header Component', () => {
  let wrapper;
  let mockLogout;
  let mockLogOutFirebase;

  beforeEach(() => {
    mockLogout = jest.fn();
    mockLogOutFirebase = jest.fn();
    wrapper = shallow(
      <Header 
        logout={mockLogout} 
        logOutFirebase={mockLogOutFirebase}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe('mapStateToProps', () => {
    it('return an object of user', () => {
      const mockState = {user: {name: 'Tim'}, logout: 'no'};
      const expected = {user: {name: 'Tim'}};
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});