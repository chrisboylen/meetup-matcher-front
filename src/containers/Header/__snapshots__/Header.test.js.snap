// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<header
  className="navbar"
>
  <Link
    className="title-link"
    replace={false}
    to="/"
  >
    <h1>
      Meetup Matcher
    </h1>
  </Link>
  <div
    className="user-links"
  >
    <Link
      className="form-link login"
      replace={false}
      to="/login"
    >
      Login
    </Link>
    <Link
      className="form-link signup"
      replace={false}
      to="/signup"
    >
      Signup
    </Link>
  </div>
</header>
`;

exports[`Header Component should match snapshot with no user 1`] = `
<header
  className="navbar"
>
  <Link
    className="title-link"
    replace={false}
    to="/"
  >
    <h1>
      Meetup Matcher
    </h1>
  </Link>
  <div
    className="user-links"
  >
    <Link
      className="form-link login"
      replace={false}
      to="/login"
    >
      Login
    </Link>
    <Link
      className="form-link signup"
      replace={false}
      to="/signup"
    >
      Signup
    </Link>
  </div>
</header>
`;
